app/main.py

```diff
@@ -1,44 +1,78 @@
+import asyncio
 import time
-
 from iot.devices import HueLightDevice, SmartSpeakerDevice, SmartToiletDevice
 from iot.message import Message, MessageType
 from iot.service import IOTService
+from typing import Any, Awaitable
+
+
+async def run_sequence(*functions: Awaitable[Any]) -> None:
+    for function in functions:
+        await function
+
+
+async def run_parallel(*functions: Awaitable[Any]) -> None:
+    await asyncio.gather(*functions)
+
+
+async def run_wake_up_program(
+        service: IOTService,
+        hue_light_id: str,
+        speaker_id: str
+) -> None:
+    await run_sequence(
+        service.send_msg(Message(hue_light_id, MessageType.SWITCH_ON)),
+        run_parallel(
+            service.send_msg(Message(speaker_id, MessageType.SWITCH_ON)),
```

Comment:
Turning speaker on and playing a song on the same speaker should be done in a sequence (not parallel).